hot-fix/generic-byIdSpecification

aaabramov opened the pull request:
- WarehouseGetByIdSpecification
- Generic GetByIdSpecification

ITsvetkoFF commented on persistence/src/main/java/edu/softserve/zoo/persistence/specification/impl/WarehouseGetByIdSpecification.java:
```diff
@@ -1,23 +1,12 @@
-    private static final String WAREHOUSE_TYPE = Warehouse.class.getSimpleName();
-    private int idToRemove;
-
+public class WarehouseGetByIdSpecification extends GetByIdSpecification<Warehouse> {
```
Why idToRemove? Can't get it! Is this get specification?

JuliaAvdeionok committed:
Added Task specification, service, controller

JuliaAvdeionok committed:
Added Task specification, service, controller

JuliaAvdeionok committed:
Merge remote-tracking branch 'origin/demo-1' into demo-1/tasks-rest-api

aaabramov committed:
+ Warehouse Controller
+ Warehouse Service
+ Warehouse Repository
+ WarehouseDTO
+ Warehouse Specifications
+ BadFormatException
+ Spring Test dependency for Service tests;+ Spring Test dependency for Service tests;
+ Generic GetByIdSpecification
+ WarehouseGetByIdSpecification

aaabramov committed:
Merge branch 'demo-1' into feature/warehouse

# Conflicts:
#	persistence/src/main/java/edu/softserve/zoo/persistence/specification/impl/WarehouseGetByIdSpecification.java
#	service/src/main/java/edu/softserve/zoo/service/impl/WarehouseServiceImpl.java
#	service/src/test/java/edu/softserve/zoo/service/WarehouseServiceTest.java
